include/SFML/Network/TcpSocket.hpp

```diff
@@ -128,6 +174,339 @@ class SFML_NETWORK_API TcpSocket : public Socket
+                             std::size_t      certificateChainSize,
+                             const std::byte* privateKeyData,
+                             std::size_t      privateKeySize,
+                             const std::byte* privateKeyPasswordData,
+                             std::size_t      privateKeyPasswordSize);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Set up transport layer security as a server
+    ///
+    /// Once the TCP connection is connected, transport layer
+    /// security can be set up.
+    ///
+    /// All the necessary cryptographic initialization will
+    /// be performed when this function is called.
+    ///
+    /// If this function is called before the TCP connection is
+    /// connected, it will return `TlsStatus::NotConnected` and
+    /// must be called again once the TCP connection is connected.
+    ///
+    /// If this function started TLS setup but could not finish
+    /// it within this call e.g. because this socket was set to
+    /// non-blocking, it will return `TlsStatus::HandshakeStarted`
+    /// and this function will have to be called repeatedly until
+    /// `TlsStatus::HandshakeComplete` is returned. If this socket
+    /// is blocking, `TlsStatus::HandshakeComplete` should be
+    /// returned within the same function call if TLS setup was
+    /// successful.
+    ///
+    /// If `TlsStatus::Error` is returned, something went wrong
+    /// with TLS setup and the connection must be disconnected.
+    /// The client must reconnect and reattempt TLS setup again.
+    ///
+    /// As a server, a certificate chain as well as a private key
+    /// must be provided.
+    ///
+    /// The certificate and private key data should be provided in
+    /// PEM format.
+    ///
+    /// If the private key is secured by a password, the password
+    /// must be provided.
+    ///
+    /// \param certificateChainData   Certificate chain data in PEM encoding
+    /// \param privateKeyData         Private key data in PEM encoding
+    /// \param privateKeyPasswordData Private key password if required
+    ///
+    /// \return TLS status code
+    ///
+    /// \see `setupTlsClient`
+    ///
+    ////////////////////////////////////////////////////////////
+    TlsStatus setupTlsServer(std::string_view certificateChainData,
+                             std::string_view privateKeyData,
+                             std::string_view privateKeyPasswordData = {});
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the name of the TLS ciphersuite currently in use
+    ///
+    /// \return TLS ciphersuite currently in use or `std::nullopt` if TLS is not set up
+    ///
+    /// \see `setupTlsClient`, `setupTlsServer`
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] std::optional<std::string> getCurrentCiphersuiteName() const;
```

Comment:
Should this be an enum instead of an string?

Reply:
In perfect world where SFML could track everything, probably, but we cipher suites can evolve in manh different ways, we'd have to constantly update it and old library version would suddenly be unable to handle new cipher suites